Source/Core/InputCommon/ControllerInterface/Pipes/Pipes.cpp

```diff
@@ -69,9 +70,9 @@ PipeDevice::PipeDevice(int fd, const std::string& name, int id) : m_fd(fd), m_na
 {
   for (const auto& tok : s_button_tokens)
   {
-    PipeInput* btn = new PipeInput("Button " + tok);
-    AddInput(btn);
-    m_buttons[tok] = btn;
+    auto btn = std::make_unique<PipeInput>("Button " + tok);
+    m_buttons[tok] = btn.get();
```

Comment:
These can likely be removed and replaced with FindInput/FindOutput from the Device interface. I have no idea why it was like this.